client/src/entrypoints/admin/page-editor.js

```diff
@@ -86,55 +86,58 @@ function InlinePanel(opts) {
     }
   };
 
-  self.formsUl = $('#' + opts.formsetPrefix + '-FORMS');
+  self.formsElt = $('#' + opts.formsetPrefix + '-FORMS');
 
-  // eslint-disable-next-line func-names
-  self.updateMoveButtonDisabledStates = function () {
+  self.updateMoveButtonDisabledStates = function updateMoveButtons() {
     if (opts.canOrder) {
-      const forms = self.formsUl.children('li:not(.deleted)');
-      // eslint-disable-next-line func-names
-      forms.each(function (i) {
-        $('ul.controls .inline-child-move-up', this)
-          .toggleClass('disabled', i === 0)
-          .toggleClass('enabled', i !== 0);
-        $('ul.controls .inline-child-move-down', this)
-          .toggleClass('disabled', i === forms.length - 1)
-          .toggleClass('enabled', i !== forms.length - 1);
+      const forms = self.formsElt.children(':not(.deleted)');
+      forms.each(function updateButtonStates(i) {
+        const isFirst = i === 0;
+        const isLast = i === forms.length - 1;
+        console.log(isFirst, isLast);
+        $('[data-inline-panel-child-move-up]', this).prop('disabled', isFirst);
+        $('[data-inline-panel-child-move-down]', this).prop('disabled', isLast);
       });
     }
   };
 
   // eslint-disable-next-line func-names
   self.updateAddButtonState = function () {
     if (opts.maxForms) {
-      const forms = $('> [data-inline-panel-child]', self.formsUl).not(
+      const forms = $('> [data-inline-panel-child]', self.formsElt).not(
         '.deleted',
       );
       const addButton = $('#' + opts.formsetPrefix + '-ADD');
 
       if (forms.length >= opts.maxForms) {
-        addButton.addClass('disabled');
+        addButton.prop('disabled', true);
       } else {
-        addButton.removeClass('disabled');
+        addButton.prop('disabled', false);
       }
     }
   };
 
   // eslint-disable-next-line func-names
   self.animateSwap = function (item1, item2) {
-    const parent = self.formsUl;
-    const children = parent.children('li:not(.deleted)');
-
-    // Apply moving class to container (ul.multiple) so it can assist absolute positioning of it's children
-    // Also set it's relatively calculated height to be an absolute one,
-    // to prevent the containercollapsing while its children go absolute
-    parent.addClass('moving').css('height', parent.height());
+    const parent = self.formsElt;
+    const children = parent.children(':not(.deleted)');
+
+    // Position children absolutely and add hard-coded height
+    // to prevent scroll jumps when reordering.
+    parent.css({
+      position: 'relative',
+      height: parent.height(),
+    });
```

Comment:
Here and below – this applies the same effects as before, just directly via inline styles rather than classes. This makes it a bit easier to understand the dependencies.